Deploy-CAM.ps1

```diff
@@ -135,6 +180,48 @@ function Login-AzureRmAccountWithBetterReporting($Credential) {
     }
 }
 
+# uses session instance profile and TokenCache and returns an access token without having to authentication a second time
+function Get-AzureRmCachedAccessToken() {
+    $ErrorActionPreference = 'Stop'
+    if(-not (Get-Module AzureRm.Profile)) {
+        Import-Module AzureRm.Profile
+    }
+    $azureRmProfileModuleVersion = (Get-Module AzureRm.Profile).Version
+    
+    # refactoring performed in AzureRm.Profile v3.0 or later
+    if($azureRmProfileModuleVersion.Major -ge 3) {
+        $azureRmProfile = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile
+        if(-not $azureRmProfile.Accounts.Count) {
+            Write-Error "Ensure you have logged in before calling this function."
+        }
+    } else {
+        # AzureRm.Profile < v3.0
+        $azureRmProfile = [Microsoft.WindowsAzure.Commands.Common.AzureRmProfileProvider]::Instance.Profile
+        if(-not $azureRmProfile.Context.Account.Count) {
+            Write-Error "Ensure you have logged in before calling this function."
+        }
+    }
+    $currentAzureContext = Get-AzureRmContext
+    $profileClient = New-Object Microsoft.Azure.Commands.ResourceManager.Common.RMProfileClient($azureRmProfile)
+    Write-Debug ("Getting access token for tenant" + $currentAzureContext.Subscription.TenantId)
+    $token = $profileClient.AcquireAccessToken($currentAzureContext.Subscription.TenantId)
+    $token.AccessToken
```

Comment:
return?

Reply:
Yeah a return at the end of a PowerShell function is just 'style' so not really needed but my C++ is showing in my preference for having it :)

Reply:
No need for return keyword.